pricewars_merchant.py

```diff
@@ -68,13 +73,32 @@ def calculate_id(token: str) -> str:
         return base64.b64encode(hashlib.sha256(token.encode('utf-8')).digest()).decode('utf-8')
 
     def run(self):
+        # initial random sleep to avoid starting merchants in sync
+        time.sleep(2 * random.random())
+
         start_time = time.time()
+        cnt = 1
         while True:
+            il = self.settings['update interval']
+            lb = self.settings['interval_lower_bound_relative']
+            ub = self.settings['interval_upper_bound_relative']
+
             if self.state == 'running':
                 self.update_offers()
-            # Waiting for the length of the update interval minus the execution time
-            time.sleep(self.settings['update interval'] -
-                       ((time.time() - start_time) % self.settings['update interval']))
+
+            # determine required sleep length for next interval
+            rdm_interval_length = (il * lb) + (random.random() * il * (ub - lb))
+            # calculate next expected update timespamp (might be in the past)
+            next_update_ts = start_time + il * (cnt-1) + rdm_interval_length
```

Comment:
I think here `next_update_ts += rdm_interval_length` is sufficient (once `next_update_ts` is initialized to `start_time`).